jetty-maven-plugin/src/main/java/org/eclipse/jetty/maven/plugin/JettyRunMojo.java

```diff
@@ -518,64 +528,41 @@ public void restartWebApp(boolean reconfigureScanner) throws Exception
         startScanner();
         getLog().info("Restart completed at "+new Date().toString());
     }
-    
-    
-    
-    
-    /**
-     * @return
-     */
-    private List<File> getDependencyFiles()
-    {
-        List<File> dependencyFiles = new ArrayList<>();
-        for ( Artifact artifact : projectArtifacts)
-        {
-            // Include runtime and compile time libraries, and possibly test libs too
-            if(artifact.getType().equals("war"))
-                continue;
-
-            if (Artifact.SCOPE_PROVIDED.equals(artifact.getScope()))
-                continue; //never add dependencies of scope=provided to the webapp's classpath (see also <useProvidedScope> param)
 
-            if (Artifact.SCOPE_TEST.equals(artifact.getScope()) && !useTestScope)
-                continue; //only add dependencies of scope=test if explicitly required
-
-            MavenProject mavenProject = getProjectReference( artifact, project );
-            if (mavenProject != null)
-            {
-                File projectPath = Paths.get(mavenProject.getBuild().getOutputDirectory()).toFile();
-                getLog().debug( "Adding project directory " + projectPath.toString() );
-                dependencyFiles.add( projectPath );
-                continue;
-            }
+    private Collection<Artifact> getWebInfLibArtifacts(Set<Artifact> artifacts)
+    {
+        return artifacts.stream()
+                .filter(this::canPutArtifactInWebInfLib)
+                .collect(Collectors.toList());
+    }
 
-            dependencyFiles.add(artifact.getFile());
-            getLog().debug( "Adding artifact " + artifact.getFile().getName() + " with scope "+artifact.getScope()+" for WEB-INF/lib " );   
+    private Collection<Artifact> getWebInfLibArtifacts(MavenProject mavenProject)
```

Comment:
I find it a bit confusing to have the same name for 2 methods

Reply:
well - regular method overloading :) I need both for further refactoring (WAR overlays)